### ISSUE TITLE
TypeError: Cannot read property 'lineMap' of undefined

### ISSUE THREAD

**Issue by Gwildor** [smrq/tsify#45]
Hi there,

I can't seem to set up tsify to work when using a very basic gulpfile. Not sure what more to say about it, so here's the stuff I have:

`gulpfile.js`
```javascript
var gulp = require('gulp');
var browserify = require('browserify');
var tsify = require('tsify');
var source = require('vinyl-source-stream');


gulp.task('compile-js', function() {
    return browserify()
        .add('main.ts')
        .plugin('tsify', {target: 'ES6'})
        .bundle()
        .pipe(source('bundle.js'));
});

gulp.task('default', ['compile-js']);
```

`main.ts`
```typescript
/// <reference path="typings/node.d.ts" />
```

The typing is copied from [https://github.com/borisyankov/DefinitelyTyped/blob/master/node/node.d.ts](DefinitelyTyped).

Stacktrace:

```
$ gulp
[15:15:07] Using gulpfile <snip>/gulpfile.js
[15:15:07] Starting 'compile-js'...

<snip>/node_modules/tsify/node_modules/typescript/bin/typescript.js:2448
        return sourceFile.lineMap || (sourceFile.lineMap = computeLineStarts(s
                         ^
TypeError: Cannot read property 'lineMap' of undefined
    at getLineStarts (<snip>/node_modules/tsify/node_modules/typescript/bin/typescript.js:2448:26)
    at Object.getLineAndCharacterOfPosition (<snip>/node_modules/tsify/node_modules/typescript/bin/typescript.js:2468:50)
    at SyntaxError.CompileError (<snip>/node_modules/tsify/lib/CompileError.js:7:15)
    at <snip>/node_modules/tsify/lib/Tsifier.js:126:22
    at Array.forEach (native)
    at Tsifier.checkSemantics (<snip>/node_modules/tsify/lib/Tsifier.js:125:22)
    at Tsifier.compile (<snip>/node_modules/tsify/lib/Tsifier.js:83:33)
    at Tsifier.generateCache (<snip>/node_modules/tsify/lib/Tsifier.js:58:7)
    at DestroyableTransform.flush [as _flush] (<snip>/node_modules/tsify/index.js:40:12)
    at DestroyableTransform.<anonymous> (<snip>/node_modules/tsify/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:135:12)
```

Note that I get a different result when I change `ES6` to `ES5` in the gulpfile:

```
$ gulp
[17:36:57] Using gulpfile <snip>/gulpfile.js
[17:36:57] Starting 'compile-js'...

events.js:72
        throw er; // Unhandled 'error' event
              ^
TypeScript error: typings/node.d.ts(198,26): Error TS2304: Cannot find name 'DataView'.
```

The missing `DataView` seems to be something that was recently changed in the typing of DefinitelyTyped. Unfortunately, I don't know enough of TypeScript yet to understand what is happening exactly.

**Comment by smrq** [comment]
Cool, thanks for the reproducing test case.  I've found at least two issues here -- one with the error output, and one with ES6 mode.  Working on it.

Re: the ES5 error, yeah, looks like a problem with the type definition: https://github.com/borisyankov/DefinitelyTyped/issues/4249

**Comment by smrq** [comment]
The issues with `tsify` specifically are fixed in v0.10.2.  Now it fails in exactly the same way that `tsc` does.  I suggest watching the above issue from DefinitelyTyped for further fixes.

Thanks for the report!
